gensim/models/phrases.py

```diff
@@ -165,7 +165,10 @@ def __str__(self):
     def learn_vocab(self, sentences):
         """Collect unigram/bigram counts from the `sentences` iterable."""
         if not self.recode_to_utf8 and sentences is not None:
-            sentence = list(next(iter(sentences)))
+            try:
+                sentence = list(next(iter(sentences)))
+            except:
+                raise ValueError("Input can not be empty list or generator.")
```

Comment:
Unless it's really a special case (and I don't think it is), there's no need to treat it in a special way.

No exception, no warning -- there's nothing special to an empty corpus with regard to Phrases, except your new check for its first element.